### PR TITLE
Remember socket floating

### PR DESCRIPTION
**Pull request by dabbler0**
A new remember-socket PR to work with multiple documents. This will track the location of a socket through all (I think) operations except setValue.

### CODE REVIEW

**Comment by davidbau** [comment]
Can the changes be squashed?

File: src/controller.coffee [hunk 1]
```diff
@@ -817,26 +820,50 @@ Editor::redo = ->
...
 
   @popRedo()
   @redrawMain()
   return
 
 Editor::undoCapture = ->
-  @pushUndo 'CAPTURE'
+  @pushUndo new CapturePoint(@rememberedSockets)
+
+class CapturePoint
```
Comment (davidbau) [inline src/controller.coffee]: OK, there's something going on here, but I don't get it.  Can you start by explaining the idea of how you're solving the problem?

Comment (dabbler0) [inline src/controller.coffee]: I'll add some comments. This was to preserve the remembered socket contents across undo and redo.

File: src/controller.coffee [hunk 2]
```diff
@@ -817,26 +822,57 @@ Editor::redo = ->
...
+  # at this point in the undo stack.
+  if @undoStack[@undoStack.length - 1] instanceof CapturePoint
+    @rememberedSockets = @undoStack[@undoStack.length - 1].rememberedSockets.map (x) -> x.clone()
 
   @popRedo()
   @redrawMain()
   return
 
+# ## undoCapture and CapturePoint ##
+# A CapturePoint is a flag indicating that the undo stack
```
Comment (davidbau) [inline src/controller.coffee]: ... "is a sentiinal in the undo stock indicating the boundary of a user operation" ...

**Comment by davidbau** [comment]
Thanks for adding the tests and comments!

LGTM.

**Pull request merged**